iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/convert.inl

```diff
@@ -86,15 +79,12 @@ inline bool convert::fromString<char>(const char* v, char& dest) noexcept
     return true;
 }
 
-/// @NOLINTJUSTIFICATION @todo iox-#260 convert must be refactored
-/// @NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
-template <>
-inline bool convert::fromString<string<100>>(const char* v, string<100>& dest) noexcept
+template <uint64_t Capacity>
+inline bool convert::fromString(const char* v, string<Capacity>& dest) noexcept
```

Comment:
I see now why we would need to add this overload. In theory it should be tested in the way I requested with a `string<100>`.

Reply:
I added a simple test for `convert::fromString` overload. Also I actually modified the implementation to return `false` if the source size exceeds the `cxx::string` capacity.